doc/Rules.md

```diff
@@ -56,7 +56,8 @@ Aliases: first-heading-h1, first-header-h1
 Parameters: level (number; default 1)
 
 > Note: *MD002 has been deprecated and is disabled by default.*
-> [MD041/first-line-heading](#md041) offers an improved implementation.
+> [MD041/first-line-heading](#md041---first-line-in-a-file-should-be-a-top-level-heading)
```

Comment:
Please keep the short form like it was here and elsewhere.

Reply:
I changed that because it's not respecting the new rule `valid-link-fragments`.
Currently, we are checking headings, not `id`, not `name` attribute from a link `<a>`.
So yes even if `#md041` is a valid fragment because of `<a name="md041"></a>`, there are no heading with `md041` as content so the rule fails.

I might also get the `id` and `name` attribute to make it working.

Reply:
Ah, yes, this is a valid scenario, so let's allow it.

Reply:
Yes, we should allow it.
I tried but could not figure out, how can I loop trough each inline HTML and check if it's a link, so we add the `name` content inside the `validFragments` variable.

Is there a utility function to do that?

Reply:
I don't think so. It might be necessary to use a RegExp which is unfortunate because that's trickier and less reliable.